src/tests/Microsoft.PowerFx.Connectors.Tests.Shared/PowerPlatformTabularTests.cs

```diff
@@ -153,10 +153,96 @@ public async Task SQL_CdpTabular_GetTables()
+
+            CdpTable connectorTable = tables.First(t => t.DisplayName == realTableName);
+
+            Assert.False(connectorTable.IsInitialized);
+            Assert.Equal(realTableName, connectorTable.DisplayName);
+
+            testConnector.SetResponseFromFiles(@"Responses\SQL GetSchema Products.json", @"Responses\SQL GetRelationships SampleDB.json");
+            await connectorTable.InitAsync(client, $"/apim/sql/{connectionId}", CancellationToken.None, logger);
+            Assert.True(connectorTable.IsInitialized);
+
+            CdpTableValue sqlTable = connectorTable.GetTableValue();
+            Assert.True(sqlTable._tabularService.IsInitialized);
+            Assert.True(sqlTable.IsDelegable);
+            Assert.Equal("*[Color:s, DiscontinuedDate:d, ListPrice:w, ModifiedDate:d, Name:s, ProductCategoryID:w, ProductID:w, ProductModelID:w, ProductNumber:s, SellEndDate:d, SellStartDate:d, Size:s, StandardCost:w, ThumbNailPhoto:o, ThumbnailPhotoFileName:s, Weight:w, rowguid:s]", sqlTable.Type._type.ToString());
+
+            HashSet<IExternalTabularDataSource> ads = sqlTable.Type._type.AssociatedDataSources;
+            Assert.NotNull(ads);
+
+            Assert.NotNull(sqlTable._connectorType);
+            Assert.Null(sqlTable._connectorType.Relationships); // TO BE CHANGED, x-ms-relationships only for now
+
+            SymbolValues symbolValues = new SymbolValues().Add(fxTableName, sqlTable);
+            RuntimeConfig rc = new RuntimeConfig(symbolValues).AddService<ConnectorLogger>(logger);
+
+            // Expression with tabular connector
+            string expr = @$"First({fxTableName}).Name";
+            CheckResult check = engine.Check(expr, options: new ParserOptions() { AllowsSideEffects = true }, symbolTable: symbolValues.SymbolTable);
+            Assert.True(check.IsSuccess);
+
+            // Use tabular connector. Internally we'll call CdpTableValue.GetRowsInternal to get the data
+            testConnector.SetResponseFromFile(@"Responses\SQL GetItems Products.json");
+            FormulaValue result = await check.GetEvaluator().EvalAsync(CancellationToken.None, rc);
+
+            StringValue address = Assert.IsType<StringValue>(result);
+            Assert.Equal("HL Road Frame - Black, 58", address.Value);
+
+            bool b = sqlTable.TabularRecordType.TryGetFieldExternalTableName("ProductModelID", out string externalTableName, out string foreignKey);
+            Assert.True(b);
+            Assert.Equal("ProductModel", externalTableName); // Display Name
+            Assert.Equal("ProductModelID", foreignKey);
+
+            testConnector.SetResponseFromFiles(@"Responses\SQL GetSchema ProductModel.json", @"Responses\SQL GetRelationships SampleDB.json");
```

Comment:
>@"Responses\SQL GetRelationships SampleDB.json" [](http://example.com/codeflow?start=93&length=47)

As there is no cache, we query SQL relationships a second time - See CdpTableResolver.cs lines 90-91